### DISCUSSION TITLE
Helpers/Documentation for traversing rich-text AST

### DISCUSSION THREAD

**Discussion by jamespohalloran** [tinacms/tinacms#3364] [Ideas]
When migrating markdown-based sites to Tina, there's a bit of a gap trying to know how to tap into the rich-text response, and format it for:
- Plain text (used for things like SEO meta values, excerpts, etc)
- Filtering the AST for things like table of contents.  

In the remark ecosystem, there are [lots of helpers](https://github.com/syntax-tree/mdast-util-toc#related) for doing this type of thing: 

We could also use some documentation for the structure of our AST.

**Comment by jeffsee55** [comment]
I think the only thing keeping us from shipping this is that the raw string needs to be up-to-date with edits in the rich-text editor. This means we need to ship our stringify logic, which we currently only do for the iframe version. If we "hide" the raw string in the markdown AST then we can ship it without a breaking change, perhaps the more correct way is to return `rich-text` as an object though, which would be a bit jarring for users who haven't moved to the iframe yet.

**Reply by jeffsee55** [reply]
Of course, this then brings up the can of worms that if you're working with the raw markdown string you'll need to load remark plugins client-side (so they're reactive) for handling updates, which is a pattern we've been able to avoid entirely up to this point. 

There's potential for us to capture the 80% use case and handle that stuff out of the box:
- Table of contents
- Raw string for SEO purposes

**Reply by jamespohalloran** [reply]
> Raw string for SEO purposes

Is this on the rich-text field level?

I know we've talked about return the full document's raw string, but for this discussion, I'm moreso looking for a defacto way of getting a rich-text field's raw string. (Even if it's just a code-snippet in our docs as an MVP)
Also, another example outside of SEO is generating an anchor tag from rich-text. I think getting the string for a rich-text field will be a pretty common requirement.

**Reply by Genyus** [reply]
Hello, I've been working with TinaCMS for a couple weeks building a POC for a larger piece of work and it's largely been working out so far, but I came here trying to figure out how I can extract the raw Markdown string from the rich-text editor. Is there any method available to do so yet?
